### PR TITLE
TMEDIA-472: Susbscription Drawer Overlay Component

### PR DESCRIPTION
**Pull request by nschubach**
## Description
_Information about what you changed for this PR_

## Jira Ticket
- [TMEDIA-472](https://arcpublishing.atlassian.net/browse/TMEDIA-472)

## Acceptance Criteria
* Display only frame of the bottom drawer
* Vertical overflow/scrolling functionality
  * starts at 50% VH and expands to 75% VH as user scrolls
* Scroll locking prevents content behind the overlay from being seen
* Opacity is applied to rest of document & prevents interaction with the content
  * Header Nav Chain is excluded from this
* Hamburger menu continues to work when bottom drawer overlay is rendered
* Mobile breakpoints & landscape view

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-472-subscription-overlay`
2. Run storybook locally: `npm run storybook`
3. Validate the overlay and drawer with: http://localhost:60001/?path=/story/blocks-subscriptions-components-subscription-overlay--simple-content

### After
image
image
image

## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
I wonder if this is something that can be easily addressed within the scope of this ticket

Screen Shot 2021-09-28 at 11 09 59

https://dequeuniversity.com/rules/axe/4.2/scrollable-region-focusable?application=axeAPI

**Review by JackHowa — CHANGES_REQUESTED** [review]
Good job on interaction with 50vh and 75vh; based on testing in sb, seems to be working as expected for mobile and desktop. Curious about this ac "Header Nav Chain is excluded from this". can you cover this in storybook? I haven't really been able to test it based on directions

File: blocks/subscriptions-block/components/SubscriptionOverlay/index.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        setScrollDelta(overlayTouchLast - event.changedTouches[0].clientY);
+        setOverlayTouchLast(event.changedTouches[0].clientY);
+      }}
+      onTouchStart={(event) => {
+        setOverlayTouchLast(event.changedTouches[0].clientY);
+      }}
+      role="alert"
+    >
+      <div
+        aria-live="assertive"
```
Comment (matthewroach) [inline blocks/subscriptions-block/components/SubscriptionOverlay/index.jsx]: There is no need for `aria-live` to be used on this element as the parent element is using `role="alert"` which provides the same behaviour.

Actually - `aria-live` would be useful if the children of this element are rendered after this element is loaded onto the DOM.

**Comment by nschubach** [comment]
> I wonder if this is something that can be easily addressed within the scope of this ticket
> 
> Screen Shot 2021-09-28 at 11 09 59
> 
> https://dequeuniversity.com/rules/axe/4.2/scrollable-region-focusable?application=axeAPI

I had an anchor element at the top of this overlay to prevent this very thing, but I was getting other complaints about using the anchor as an anchor and not a link.  I decided to remove that in favor of letting the content that will be rendered become the focusable content.  So, this error will go away when there's a button/field in the overlay that the user can interact with.  In hindsight, I probably could have done that for the examples.

**Comment by nschubach** [comment]
> Good job on interaction with 50vh and 75vh; based on testing in sb, seems to be working as expected for mobile and desktop. Curious about this ac "Header Nav Chain is excluded from this". can you cover this in storybook? I haven't really been able to test it based on directions

Any element placed into the page that is positioned absolute will fill the body with the exception of the menu currently.  I didn't know how to show this either in storybook, but in testing this it will show up under the header when placed in a block.

File: blocks/subscriptions-block/components/SubscriptionOverlay/index.jsx (hunk 1 above)
Comment (nschubach) [reply blocks/subscriptions-block/components/SubscriptionOverlay/index.jsx]: Removed `aria-live`

**Review by JackHowa — APPROVED** [review]
Good work Nick. I like the use of storybook and a clear ac

**Pull request merged**